--- NetworkManager-0.8.2/network-manager-applet-0.8.2/src/applet.c	2010-11-02 21:28:16.850180001 -0400
+++ NetworkManager-0.8.2.hacked/network-manager-applet-0.8.2/src/applet.c	2010-11-02 20:58:40.797180001 -0400
@@ -2736,7 +2736,7 @@
 	return *icon;
 }
 
-#define FALLBACK_ICON_NAME "gtk-dialog-error"
+#define FALLBACK_ICON_NAME "dialog-error"
 
 static gboolean
 nma_icons_reload (NMApplet *applet)
