commit f53171cc0f0e38c16a6d7b0f50ed2e545b9ec1df
Author: Dan Williams <dcbw@redhat.com>
Date:   Fri Nov 5 14:58:51 2010 -0500

    trivial: fix for GTK 3.0 (bgo #632328)

diff --git a/src/applet.c b/src/applet.c
index 90f606c..7e13e2a 100644
--- a/src/applet.c
+++ b/src/applet.c
@@ -2936,7 +2936,11 @@ applet_pre_keyring_callback (gpointer user_data)
 	if (applet->menu)
 		window = gtk_widget_get_window (applet->menu);
 	if (window) {
+#if GTK_CHECK_VERSION(2,23,0)
+		screen = gdk_window_get_screen (window);
+#else
 		screen = gdk_drawable_get_screen (window);
+#endif
 		display = gdk_screen_get_display (screen);
 		g_object_ref (display);
 
@@ -2957,7 +2961,11 @@ applet_pre_keyring_callback (gpointer user_data)
 	if (applet->context_menu)
 		window = gtk_widget_get_window (applet->context_menu);
 	if (window) {
+#if GTK_CHECK_VERSION(2,23,0)
+		screen = gdk_window_get_screen (window);
+#else
 		screen = gdk_drawable_get_screen (window);
+#endif
 		display = gdk_screen_get_display (screen);
 		g_object_ref (display);
 
