diff -up a/src/applet.c.gtk3-6 b/src/applet.c
--- a/src/applet.c.gtk3-6	2010-11-17 18:57:59.150709005 -0600
+++ b/src/applet.c	2010-11-17 18:58:03.384709001 -0600
@@ -515,7 +515,7 @@ menu_title_item_expose (GtkWidget *widge
 #else
 	allocation = widget->allocation;
 #endif
-	cairo_translate (cr, widget->allocation.x, widget->allocation.y);
+	cairo_translate (cr, allocation.x, allocation.y);
 
 	text = gtk_label_get_text (GTK_LABEL (label));
 
