commit 06c7c14a5e3c6eaf9975c97e15f56bf7ca5f5c75
Author: Dan Williams <dcbw@redhat.com>
Date:   Sat Nov 6 13:31:13 2010 -0500

    mobile: more gtk3 fixes (bgo #632652)

diff --git a/src/utils/mobile-wizard.c b/src/utils/mobile-wizard.c
index 8445132..d562fc5 100644
--- a/src/utils/mobile-wizard.c
+++ b/src/utils/mobile-wizard.c
@@ -747,11 +747,23 @@ providers_setup (MobileWizard *self)
 	gtk_table_attach (GTK_TABLE (unlisted_table), alignment,
 	                  1, 2, 0, 1, GTK_EXPAND | GTK_FILL, 0, 6, 6);
 
+#if GTK_CHECK_VERSION(2,23,0)
+	self->provider_unlisted_type_combo = gtk_combo_box_text_new ();
+#else
 	self->provider_unlisted_type_combo = gtk_combo_box_new_text ();
+#endif
 	gtk_label_set_mnemonic_widget (GTK_LABEL (label), self->provider_unlisted_type_combo);
+#if GTK_CHECK_VERSION(2,23,0)
+	gtk_combo_box_text_append_text (GTK_COMBO_BOX_TEXT (self->provider_unlisted_type_combo),
+#else
 	gtk_combo_box_append_text (GTK_COMBO_BOX (self->provider_unlisted_type_combo),
+#endif
 	                           _("My provider uses GSM technology (GPRS, EDGE, UMTS, HSPA)"));
+#if GTK_CHECK_VERSION(2,23,0)
+	gtk_combo_box_text_append_text (GTK_COMBO_BOX_TEXT (self->provider_unlisted_type_combo),
+#else
 	gtk_combo_box_append_text (GTK_COMBO_BOX (self->provider_unlisted_type_combo),
+#endif
 	                           _("My provider uses CDMA technology (1xRTT, EVDO)"));
 	gtk_combo_box_set_active (GTK_COMBO_BOX (self->provider_unlisted_type_combo), 0);
 
