--- NetworkManager-0.7.0/libnm-util/Makefile.am.fpic	2007-08-30 21:17:33.000000000 -0400
+++ NetworkManager-0.7.0/libnm-util/Makefile.am	2007-08-30 21:18:12.000000000 -0400
@@ -5,6 +5,7 @@ lib_LTLIBRARIES=libnm-util.la
 libnm_util_la_CPPFLAGS =				\
 	$(GLIB_CFLAGS)					\
 	$(DBUS_CFLAGS)					\
+	-fPIC		\
 	-DDBUS_API_SUBJECT_TO_CHANGE		\
 	-DG_DISABLE_DEPRECATED			\
 	-DGDK_DISABLE_DEPRECATED			\
--- NetworkManager-0.7.0/src/ppp-manager/Makefile.am.fpic	2007-08-30 14:45:45.000000000 -0400
+++ NetworkManager-0.7.0/src/ppp-manager/Makefile.am	2007-08-30 19:43:20.000000000 -0400
@@ -33,7 +33,7 @@ nm_pppd_plugin_so_SOURCES = \
 	nm-pppd-plugin.h	\
 	nm-ppp-status.h
 
-nm_pppd_plugin_so_CPPFLAGS = $(DBUS_CFLAGS)	$(GLIB_CFLAGS)
+nm_pppd_plugin_so_CPPFLAGS = $(DBUS_CFLAGS)	$(GLIB_CFLAGS) -fPIC
 
 nm_pppd_plugin_so_LDFLAGS =	-shared
 nm_pppd_plugin_so_LDADD = \
