--- NetworkManager-0.6.5/nm-applet-0.6.5/src/Makefile.am.buildfix	2007-02-02 10:53:26.000000000 -0500
+++ NetworkManager-0.6.5/nm-applet-0.6.5/src/Makefile.am	2007-03-15 14:20:32.000000000 -0400
@@ -2,6 +2,9 @@
 
 bin_PROGRAMS = nm-applet
 
+INCLUDES = -I${top_srcdir}/../include \
+           -I${top_srcdir}/../libnm-util
+
 nm_applet_CPPFLAGS =									\
 	$(DBUS_CFLAGS)										\
 	$(GTHREAD_CFLAGS)									\
@@ -12,7 +15,6 @@
 	$(GCONF_CFLAGS)									\
 	$(LIBGNOMEUI_CFLAGS)								\
 	$(GNOME_KEYRING_CFLAGS)								\
-	$(NM_CFLAGS)									\
 	-DICONDIR=\""$(datadir)/icons"\"						\
 	-DGLADEDIR=\""$(gladedir)"\"							\
 	-DBINDIR=\""$(bindir)"\"								\
@@ -112,7 +114,7 @@
 	$(GCONF_LIBS)						\
 	$(LIBGNOMEUI_LIBS)					\
 	$(GNOME_KEYRING_LIBS)				\
-	$(NM_LIBS) \
+	-L${top_srcdir}/../libnm-util  $(top_builddir)/../libnm-util/libnm-util.la \
 	$(NULL)
 
 if WITH_NOTIFY
--- NetworkManager-0.6.5/nm-applet-0.6.5/configure.ac.buildfix	2007-03-15 11:51:49.000000000 -0400
+++ NetworkManager-0.6.5/nm-applet-0.6.5/configure.ac	2007-03-15 16:33:34.000000000 -0400
@@ -18,10 +18,10 @@
 dnl This doesn't depend on gnome support, only on gnome-common, autogen already
 dnl uses gnome-autogen
 dnl
-GNOME_COMMON_INIT
-GNOME_DEBUG_CHECK
-GNOME_COMPILE_WARNINGS([maximum])
-GNOME_MAINTAINER_MODE_DEFINES
+dnl GNOME_COMMON_INIT
+dnl GNOME_DEBUG_CHECK
+dnl GNOME_COMPILE_WARNINGS([maximum])
+dnl GNOME_MAINTAINER_MODE_DEFINES
 
 dnl
 dnl Required headers
@@ -112,10 +112,6 @@
 AC_SUBST(GOBJECT_CFLAGS)
 AC_SUBST(GOBJECT_LIBS)
 
-PKG_CHECK_MODULES(NM, NetworkManager)
-AC_SUBST(NM_CFLAGS)
-AC_SUBST(NM_LIBS)
-
 AC_ARG_WITH(notify, AC_HELP_STRING([--with-notify], [Use libnotify]), enable_notify=$withval, enable_notify=auto)
 
 PKG_CHECK_MODULES(GTK, gtk+-2.0)
