--- NetworkManager-0.7.0/nm-applet-0.7.0/configure.ac.buildfix	2007-08-30 15:04:50.000000000 -0400
+++ NetworkManager-0.7.0/nm-applet-0.7.0/configure.ac	2007-08-30 15:19:48.000000000 -0400
@@ -18,10 +18,10 @@ dnl
 dnl This doesn't depend on gnome support, only on gnome-common, autogen already
 dnl uses gnome-autogen
 dnl
-GNOME_COMMON_INIT
-GNOME_DEBUG_CHECK
-GNOME_COMPILE_WARNINGS([maximum])
-GNOME_MAINTAINER_MODE_DEFINES
+dnl GNOME_COMMON_INIT
+dnl GNOME_DEBUG_CHECK
+dnl GNOME_COMPILE_WARNINGS([maximum])
+dnl GNOME_MAINTAINER_MODE_DEFINES
 
 dnl
 dnl Required headers
@@ -62,7 +62,6 @@ AC_CHECK_LIB(nm-util, ieee_802_11_cipher
 PKG_CHECK_MODULES(NMA,
 		[dbus-glib-1 >= 0.72
 		 glib-2.0 >= 2.10
-		 NetworkManager >= 0.7.0
 		 libnm_glib
 		 gtk+-2.0 >= 2.6
 		 libglade-2.0
--- NetworkManager-0.7.0/nm-applet-0.7.0/src/Makefile.am.buildfix	2007-08-30 15:02:27.000000000 -0400
+++ NetworkManager-0.7.0/nm-applet-0.7.0/src/Makefile.am	2007-08-30 15:24:08.000000000 -0400
@@ -4,6 +4,10 @@ NULL=
 
 bin_PROGRAMS = nm-applet
 
+INCLUDES = -I${top_srcdir}/../include \
+           -I${top_srcdir}/../libnm-util \
+           -I${top_srcdir}/../libnm-glib
+
 nm_applet_CPPFLAGS =									\
 	$(NMA_CFLAGS)										\
 	-DICONDIR=\""$(datadir)/icons"\"						\
@@ -89,7 +93,9 @@ if WITH_NOTIFY
 nm_applet_SOURCES += applet-notifications.c applet-notifications.h
 endif
 
-nm_applet_LDADD = $(NMA_LIBS)
+nm_applet_LDADD = $(NMA_LIBS) \
+	-L${top_srcdir}/../libnm-util $(top_builddir)/../libnm-util/libnm-util.la \
+	-L${top_srcdir}/../libnm-glib $(top_builddir)/../libnm-glib/libnm_glib.la
 
 if WITH_NOTIFY
 nm_applet_LDADD += $(NOTIFY_LIBS)
