Index: src/nm-device-802-11-wireless.c
===================================================================
--- src/nm-device-802-11-wireless.c	(revision 2577)
+++ src/nm-device-802-11-wireless.c	(working copy)
@@ -2981,6 +2981,9 @@
 		return NM_ACT_STAGE_RETURN_POSTPONE;
 	}
 
+	/* Some cards are dumb.  Wait a second */
+	sleep (1);
+
 	iface = nm_device_get_iface (dev);
 	if (!supplicant_exec (self))
 	{
