--- NetworkManager-0.6.5/src/nm-device-802-11-wireless.c.debug	2007-04-09 11:00:46.000000000 -0400
+++ NetworkManager-0.6.5/src/nm-device-802-11-wireless.c	2007-04-09 11:02:00.000000000 -0400
@@ -2715,14 +2715,17 @@
 supplicant_exec (NMDevice80211Wireless *self)
 {
 	gboolean	success = FALSE;
-	char *	argv[4];
+	char *	argv[7];
 	GError *	error = NULL;
 	GPid		pid = -1;
 
 	argv[0] = WPA_SUPPLICANT_BIN;
 	argv[1] = "-g";
 	argv[2] = WPA_SUPPLICANT_GLOBAL_SOCKET;
-	argv[3] = NULL;
+	argv[3] = "-dd";
+	argv[4] = "-o";
+	argv[5] = "daemon";
+	argv[6] = NULL;
 
 	success = g_spawn_async ("/", argv, NULL, 0, &supplicant_child_setup, NULL,
 	                         &pid, &error);
