Index: NetworkManager-0.6.5/src/NetworkManager.c
===================================================================
--- NetworkManager-0.6.5/src/NetworkManager.c	(revision 2617)
+++ NetworkManager-0.6.5/src/NetworkManager.c	(working copy)
@@ -825,6 +825,10 @@
 		nm_add_initial_killswitch_devices (data);
 		nm_add_initial_devices (data);
 	}
+
+	/* If there weren't any killswitches, mark hardware RF to on */
+	if (g_slist_length (data->killswitch_list) == 0)
+		data->hw_rf_enabled = TRUE;
 }
 
 
