--- NetworkManager-0.6.2/initscript/RedHat/NetworkManager.in.dhcdbd	2006-03-30 12:42:28.000000000 -0500
+++ NetworkManager-0.6.2/initscript/RedHat/NetworkManager.in	2006-03-30 12:43:25.000000000 -0500
@@ -36,6 +36,11 @@
 {
 	echo $"Setting network parameters... "
 	sysctl -e -p /etc/sysctl.conf >/dev/null 2>&1
+
+	if [ ! -e /var/lock/subsys/dhcdbd ]; then
+		service dhcdbd start
+	fi
+
 	echo -n $"Starting NetworkManager daemon: "
 	daemon --check $servicename $processname --pid-file=$pidfile
 	RETVAL=$?
