diff -up NetworkManager-0.7.0/nm-applet-0.7.0/src/connection-editor/nm-connection-list.c.disable-stuff NetworkManager-0.7.0/nm-applet-0.7.0/src/connection-editor/nm-connection-list.c
--- NetworkManager-0.7.0/nm-applet-0.7.0/src/connection-editor/nm-connection-list.c.disable-stuff	2008-02-14 12:23:13.000000000 -0500
+++ NetworkManager-0.7.0/nm-applet-0.7.0/src/connection-editor/nm-connection-list.c	2008-02-14 14:47:23.000000000 -0500
@@ -218,6 +218,8 @@ list_selection_changed_cb (GtkTreeSelect
 	GtkTreeIter iter;
 	GtkTreeModel *model;
 
+return;
+
 	if (gtk_tree_selection_get_selected (selection, &model, &iter))
 		gtk_widget_set_sensitive (GTK_WIDGET (button), TRUE);
 	else
@@ -413,9 +415,11 @@ new_connection_list (NMConnectionList *l
 	                                      COL_TIMESTAMP, GTK_SORT_DESCENDING);
 	gtk_tree_view_set_model (GTK_TREE_VIEW (clist), GTK_TREE_MODEL (sort_model));
 
+if (0) {
 	g_signal_connect (G_OBJECT (clist),
 	                  "row-activated", G_CALLBACK (connection_double_clicked_cb),
 	                  NULL);
+}
 
 	gtk_tree_view_insert_column_with_attributes (GTK_TREE_VIEW (clist),
 	                                             -1, "Name", gtk_cell_renderer_text_new (),
@@ -440,6 +444,7 @@ new_connection_list (NMConnectionList *l
 	g_object_set_data (G_OBJECT (button), "nm-connection-list", list);
 	g_signal_connect (G_OBJECT (button), "clicked", G_CALLBACK (add_connection_cb), clist);
 	g_free (name);
+	gtk_widget_set_sensitive (button, FALSE);
 
 	name = g_strdup_printf ("%s_edit", prefix);
 	button = glade_xml_get_widget (list->gui, name);
@@ -449,6 +454,7 @@ new_connection_list (NMConnectionList *l
 	                  "changed", G_CALLBACK (list_selection_changed_cb),
 	                  button);
 	g_free (name);
+	gtk_widget_set_sensitive (button, FALSE);
 
 	name = g_strdup_printf ("%s_delete", prefix);
 	button = glade_xml_get_widget (list->gui, name);
@@ -458,6 +464,7 @@ new_connection_list (NMConnectionList *l
 	                  "changed", G_CALLBACK (list_selection_changed_cb),
 	                  button);
 	g_free (name);
+	gtk_widget_set_sensitive (button, FALSE);
 
 	return clist;
 }
