--- NetworkManager-0.7.0/nm-applet-0.7.0/src/applet.h.fixes	2007-08-30 21:24:02.000000000 -0400
+++ NetworkManager-0.7.0/nm-applet-0.7.0/src/applet.h	2007-08-30 21:24:12.000000000 -0400
@@ -26,6 +26,8 @@
 #include <config.h>
 #endif
 
+#include <string.h>
+
 #include <gtk/gtk.h>
 #include <gconf/gconf-client.h>
 #include <glade/glade.h>
--- NetworkManager-0.7.0/nm-applet-0.7.0/src/menu-items.c.fixes	2007-08-30 21:21:14.000000000 -0400
+++ NetworkManager-0.7.0/nm-applet-0.7.0/src/menu-items.c	2007-08-30 21:21:57.000000000 -0400
@@ -228,7 +228,7 @@ nm_network_menu_item_set_ssid (NMNetwork
 guint32
 nm_network_menu_item_get_strength (NMNetworkMenuItem * item)
 {
-	g_return_if_fail (item != NULL);
+	g_return_if_fail (item != NULL, 0);
 
 	return item->int_strength;
 }
@@ -249,8 +249,8 @@ const guchar *
 nm_network_menu_item_get_hash (NMNetworkMenuItem * item,
                                guint32 * length)
 {
-	g_return_if_fail (item != NULL);
-	g_return_if_fail (length != NULL);
+	g_return_if_fail (item != NULL, NULL);
+	g_return_if_fail (length != NULL, NULL);
 
 	*length = item->hash_len;
 	return item->hash;
