diff -up NetworkManager-0.7.0/nm-applet-0.7.0/src/connection-editor/nm-connection-list.c.disable-stuff NetworkManager-0.7.0/nm-applet-0.7.0/src/connection-editor/nm-connection-list.c
--- NetworkManager-0.7.0/nm-applet-0.7.0/src/connection-editor/nm-connection-list.c.disable-stuff	2008-02-27 11:02:34.000000000 -0500
+++ NetworkManager-0.7.0/nm-applet-0.7.0/src/connection-editor/nm-connection-list.c	2008-02-29 14:13:13.000000000 -0500
@@ -417,9 +417,11 @@ new_connection_list (NMConnectionList *l
 	                                      COL_TIMESTAMP, GTK_SORT_DESCENDING);
 	gtk_tree_view_set_model (GTK_TREE_VIEW (clist), GTK_TREE_MODEL (sort_model));
 
+if (FALSE) {
 	g_signal_connect (G_OBJECT (clist),
 	                  "row-activated", G_CALLBACK (connection_double_clicked_cb),
 	                  NULL);
+}
 
 	gtk_tree_view_insert_column_with_attributes (GTK_TREE_VIEW (clist),
 	                                             -1, "Name", gtk_cell_renderer_text_new (),
@@ -444,15 +446,19 @@ new_connection_list (NMConnectionList *l
 	g_object_set_data (G_OBJECT (button), "nm-connection-list", list);
 	g_signal_connect (G_OBJECT (button), "clicked", G_CALLBACK (add_connection_cb), clist);
 	g_free (name);
+	gtk_widget_set_sensitive (button, FALSE);
 
 	name = g_strdup_printf ("%s_edit", prefix);
 	button = glade_xml_get_widget (list->gui, name);
 	g_object_set_data (G_OBJECT (button), "nm-connection-list", list);
 	g_signal_connect (G_OBJECT (button), "clicked", G_CALLBACK (edit_connection_cb), clist);
+#if 0
 	g_signal_connect (G_OBJECT (select),
 	                  "changed", G_CALLBACK (list_selection_changed_cb),
 	                  button);
+#endif
 	g_free (name);
+	gtk_widget_set_sensitive (button, FALSE);
 
 	name = g_strdup_printf ("%s_delete", prefix);
 	button = glade_xml_get_widget (list->gui, name);
