diff -up NetworkManager-0.7.0/nm-applet-0.7.0/src/applet.c.disable-stuff NetworkManager-0.7.0/nm-applet-0.7.0/src/applet.c
--- NetworkManager-0.7.0/nm-applet-0.7.0/src/applet.c.disable-stuff	2007-12-04 14:03:14.000000000 -0500
+++ NetworkManager-0.7.0/nm-applet-0.7.0/src/applet.c	2007-12-04 14:45:32.000000000 -0500
@@ -315,6 +315,7 @@ nma_show_info_cb (GtkMenuItem *mi, NMApp
 	gtk_window_present (GTK_WINDOW (dialog));
 }
 
+#if 0
 static void
 nma_edit_connections_cb (GtkMenuItem *mi, NMApplet *applet)
 {
@@ -325,6 +326,7 @@ nma_edit_connections_cb (GtkMenuItem *mi
 
 	g_object_unref (connection_list);
 }
+#endif
 
 static void about_dialog_activate_link_cb (GtkAboutDialog *about,
                                            const gchar *url,
@@ -1442,6 +1444,7 @@ nma_menu_add_create_network_item (GtkWid
 	label = gtk_label_new_with_mnemonic (_("Create _New Wireless Network..."));
 	gtk_misc_set_alignment (GTK_MISC (label), 0.0, 0.5);
 	gtk_container_add (GTK_CONTAINER (menu_item), label);
+	gtk_widget_set_sensitive (menu_item, FALSE);
 	gtk_widget_show_all (menu_item);
 	gtk_menu_shell_append (GTK_MENU_SHELL (menu), menu_item);
 	g_signal_connect (menu_item, "activate", G_CALLBACK (new_network_item_selected), applet);
@@ -2060,6 +2063,7 @@ static GtkWidget *nma_context_menu_creat
 	gtk_image_menu_item_set_image (GTK_IMAGE_MENU_ITEM (applet->info_menu_item), image);
 	gtk_menu_shell_append (menu, applet->info_menu_item);
 
+#if 0
 	/* 'Edit Connections...' item */
 	applet->connections_menu_item = gtk_menu_item_new_with_mnemonic (_("Edit Connections..."));
 	g_signal_connect (applet->connections_menu_item,
@@ -2067,6 +2071,7 @@ static GtkWidget *nma_context_menu_creat
 				   G_CALLBACK (nma_edit_connections_cb),
 				   applet);
 	gtk_menu_shell_append (menu, applet->connections_menu_item);
+#endif
 
 	/* Separator */
 	nma_menu_add_separator_item (menu);
