diff -up NetworkManager-0.7.0/nm-applet-0.7.0/src/connection-editor/nm-connection-list.c.nochange NetworkManager-0.7.0/nm-applet-0.7.0/src/connection-editor/nm-connection-list.c
--- NetworkManager-0.7.0/nm-applet-0.7.0/src/connection-editor/nm-connection-list.c.nochange	2008-01-21 22:46:28.000000000 -0500
+++ NetworkManager-0.7.0/nm-applet-0.7.0/src/connection-editor/nm-connection-list.c	2008-01-21 22:48:00.000000000 -0500
@@ -229,7 +229,7 @@ list_selection_changed_cb (GtkTreeSelect
 	GtkTreeModel *model;
 
 	if (gtk_tree_selection_get_selected (selection, &model, &iter)) {
-		gtk_widget_set_sensitive (GTK_WIDGET (list->edit_button), TRUE);
+		gtk_widget_set_sensitive (GTK_WIDGET (list->edit_button), FALSE);
 		gtk_widget_set_sensitive (GTK_WIDGET (list->delete_button), TRUE);
 	} else {
 		gtk_widget_set_sensitive (GTK_WIDGET (list->edit_button), FALSE);
@@ -280,9 +280,11 @@ nm_connection_list_init (NMConnectionLis
 	/* buttons */
 	list->add_button = glade_xml_get_widget (list->gui, "add_connection_button");
 	g_signal_connect (G_OBJECT (list->add_button), "clicked", G_CALLBACK (add_connection_cb), list);
+	gtk_widget_set_sensitive (list->add_button, FALSE);
 
 	list->edit_button = glade_xml_get_widget (list->gui, "edit_connection_button");
 	g_signal_connect (G_OBJECT (list->edit_button), "clicked", G_CALLBACK (edit_connection_cb), list);
+	gtk_widget_set_sensitive (list->edit_button, FALSE);
 
 	list->delete_button = glade_xml_get_widget (list->gui, "delete_connection_button");
 	g_signal_connect (G_OBJECT (list->delete_button), "clicked", G_CALLBACK (delete_connection_cb), list);
