diff -up NetworkManager-0.7.0/nm-applet-0.7.0/src/connection-editor/nm-connection-list.c.foo NetworkManager-0.7.0/nm-applet-0.7.0/src/connection-editor/nm-connection-list.c
--- NetworkManager-0.7.0/nm-applet-0.7.0/src/connection-editor/nm-connection-list.c.foo	2008-02-07 16:04:38.000000000 -0500
+++ NetworkManager-0.7.0/nm-applet-0.7.0/src/connection-editor/nm-connection-list.c	2008-02-07 16:05:11.000000000 -0500
@@ -331,9 +331,11 @@ nm_connection_list_init (NMConnectionLis
 	/* buttons */
 	list->add_button = glade_xml_get_widget (list->gui, "add_connection_button");
 	g_signal_connect (G_OBJECT (list->add_button), "clicked", G_CALLBACK (add_connection_cb), list);
+	gtk_widget_set_sensitive (list->add_button, FALSE);
 
 	list->edit_button = glade_xml_get_widget (list->gui, "edit_connection_button");
 	g_signal_connect (G_OBJECT (list->edit_button), "clicked", G_CALLBACK (edit_connection_cb), list);
+	gtk_widget_set_sensitive (list->edit_button, FALSE);
 
 	list->delete_button = glade_xml_get_widget (list->gui, "delete_connection_button");
 	g_signal_connect (G_OBJECT (list->delete_button), "clicked", G_CALLBACK (delete_connection_cb), list);
