diff -up NetworkManager-0.7.0/configure.in.foo NetworkManager-0.7.0/configure.in
--- NetworkManager-0.7.0/configure.in.foo	2008-02-07 22:12:04.000000000 -0500
+++ NetworkManager-0.7.0/configure.in	2008-02-07 22:12:57.000000000 -0500
@@ -238,7 +238,8 @@ if test "$GCC" = "yes" -a "$set_more_war
 
 	for option in -Wshadow -Wmissing-declarations -Wmissing-prototypes \
 		      -Wdeclaration-after-statement -Wstrict-prototypes \
-		      -Wfloat-equal -Wno-unused-parameter -Wno-sign-compare; do
+		      -Wfloat-equal -Wno-unused-parameter -Wno-sign-compare \
+		      -fno-strict-aliasing; do
 		SAVE_CFLAGS="$CFLAGS"
 		CFLAGS="$CFLAGS $option"
 		AC_MSG_CHECKING([whether gcc understands $option])
