--- NetworkManager-0.8.1/src/system-settings/nm-sysconfig-settings.c	2010-06-30 14:55:33.000000000 -0400
+++ NetworkManager-0.8.1.hacked/src/system-settings/nm-sysconfig-settings.c	2010-11-02 14:58:13.141180011 -0400
@@ -1508,7 +1508,7 @@
 
 	priv->connections = g_hash_table_new_full (g_direct_hash, g_direct_equal, g_object_unref, NULL);
 
-	priv->authority = polkit_authority_get ();
+	priv->authority = polkit_authority_get_sync (NULL, NULL);
 	if (priv->authority) {
 		priv->auth_changed_id = g_signal_connect (priv->authority,
 		                                          "changed",
--- NetworkManager-0.8.1/src/system-settings/nm-sysconfig-connection.c	2010-07-21 19:01:08.000000000 -0400
+++ NetworkManager-0.8.1.hacked/src/system-settings/nm-sysconfig-connection.c	2010-11-02 14:58:35.397180077 -0400
@@ -613,7 +613,7 @@
 {
 	NMSysconfigConnectionPrivate *priv = NM_SYSCONFIG_CONNECTION_GET_PRIVATE (self);
 
-	priv->authority = polkit_authority_get ();
+	priv->authority = polkit_authority_get_sync (NULL, NULL);
 	if (!priv->authority) {
 		nm_log_err (LOGD_SYS_SET, "%s: error creating PolicyKit authority");
 	}
