--- NetworkManager-0.9.8.1/gtk-doc.make	2013-04-30 15:16:13.049980141 -0500
+++ NetworkManager-0.9.8.1/gtk-doc.make	2013-04-30 15:17:41.525874040 -0500
@@ -108,7 +108,7 @@ scan-build.stamp: $(HFILE_GLOB) $(CFILE_
 	            scanobj_options="--verbose"; \
 	        fi; \
 	    fi; \
-	    CC="$(GTKDOC_CC)" LD="$(GTKDOC_LD)" RUN="$(GTKDOC_RUN)" CFLAGS="$(GTKDOC_CFLAGS) $(CFLAGS)" LDFLAGS="$(GTKDOC_LIBS) $(LDFLAGS)" \
+	    CC="$(GTKDOC_CC)" LD="$(GTKDOC_LD)" RUN="$(GTKDOC_RUN)" CFLAGS="$(GTKDOC_CFLAGS)" LDFLAGS="$(GTKDOC_LIBS)" \
 	    gtkdoc-scangobj $(SCANGOBJ_OPTIONS) $$scanobj_options --module=$(DOC_MODULE); \
 	else \
 	    for i in $(SCANOBJ_FILES) ; do \
