diff -up NetworkManager-0.7.0/initscript/RedHat/NetworkManager.in.stop-later NetworkManager-0.7.0/initscript/RedHat/NetworkManager.in
--- NetworkManager-0.7.0/initscript/RedHat/NetworkManager.in.stop-later	2008-06-04 17:18:53.000000000 -0400
+++ NetworkManager-0.7.0/initscript/RedHat/NetworkManager.in	2008-06-04 17:34:32.000000000 -0400
@@ -2,7 +2,7 @@
 #
 # NetworkManager:   NetworkManager daemon
 #
-# chkconfig: - 27 73
+# chkconfig: - 27 84
 # description:  This is a daemon for automatically switching network \
 #               connections to the best available connection.
 #
@@ -12,7 +12,7 @@
 ### BEGIN INIT INFO
 # Provides: network_manager $network
 # Required-Start: messagebus haldaemon
-# Required-Stop: messagebus haldaemon
+# Required-Stop: messagebus
 # Default-Start: 2 3 4 5
 # Default-Stop: 0 1 6
 # Short-Description: start and stop NetworkManager
