diff -up NetworkManager-0.7.0/src/supplicant-manager/nm-supplicant-settings-verify.c.foo NetworkManager-0.7.0/src/supplicant-manager/nm-supplicant-settings-verify.c
--- NetworkManager-0.7.0/src/supplicant-manager/nm-supplicant-settings-verify.c.foo	2008-09-04 13:33:24.000000000 -0400
+++ NetworkManager-0.7.0/src/supplicant-manager/nm-supplicant-settings-verify.c	2008-09-04 13:33:32.000000000 -0400
@@ -66,7 +66,7 @@ static const struct validate_entry valid
 const char * pairwise_allowed[] = { "CCMP", "TKIP", "NONE", NULL };
 const char * group_allowed[] =    { "CCMP", "TKIP", "WEP104", "WEP40", NULL };
 const char * proto_allowed[] =    { "WPA", "RSN", NULL };
-const char * key_mgmt_allowed[] = { "WPA-PSK", "WPA-EAP", "IEEE8021X",
+const char * key_mgmt_allowed[] = { "WPA-PSK", "WPA-EAP", "IEEE8021X", "WPA-NONE",
                                     "NONE", NULL };
 const char * auth_alg_allowed[] = { "OPEN", "SHARED", "LEAP", NULL };
 const char * eap_allowed[] =      { "LEAP", "MD5", "TLS", "PEAP", "TTLS", "SIM",
